src/Features/Core/Portable/UseThrowExpression/AbstractUseThrowExpressionDiagnosticAnalyzer.cs

```diff
@@ -189,6 +195,18 @@ private void AnalyzeOperation(OperationAnalysisContext context, INamedTypeSymbol
             }
         }
 
+        private bool IsOnlyStatementOfIf(IIfStatement ifOperation, IThrowStatement throwStatement)
+        {
+            if (ifOperation.IfTrueStatement.Syntax == throwStatement.Syntax)
+            {
+                return true;
+            }
+
+            return ifOperation.IfTrueStatement is IBlockStatement block &&
+                   block.Statements.Length == 1 &&
+                   block.Statements[0].Syntax == throwStatement.Syntax;
```

Comment:
:bulb: I believe the first and last subexpressions are redundant (always true). The full expression could be reduced to:

```csharp
return ((IBlockStatement)ifOperation.IfTrueStatement).Statements.Length == 1;
```